criu/img-remote.c

```diff
@@ -181,6 +179,56 @@ static int setup_UNIX_client_socket(char *path)
 	return sockfd;
 }
 
+/*
+ * Using a pipe for image file transfers instead of the UNIX socket can
+ * greatly improve performance. Data can be further spliced() by the proxy
+ * to transfer data with minimal overhead.
+ * Transfer rates of up to 15GB/s can be seen with this technique.
+ */
+#define SEND_FD_FOR_READ	0
+#define SEND_FD_FOR_WRITE	1
+#define RECV_FD			2
+static int upgrade_to_pipe(int sockfd, int mode)
+{
+	return sockfd;
+
+	/*
+	 * If the other end of the pipe closes, the kernel will want to kill
+	 * us with a SIGPIPE. These signal must be ignored.
+	 */
+
+	if (mode == RECV_FD) {
+		int ret = recv_fd(sockfd);
+		close(sockfd);
+		return ret;
```

Comment:
`int ret = -1;` could be declared before the if-statement, then the following common part can be moved at the end of the function.
```C
		close(sockfd);
		return ret;
```

Reply:
Very well, I pushed your suggestions.